### PR TITLE
core/ts-version_issue_408_switch_to_yarn

### PR DESCRIPTION
**Pull request by GreatGaryCooper**
## What does this MR do?

- [x] It migrates from npm to yarn v3 (berry)
- [x] It cleans up the `package.json` scripts
- [x] Add Husky and pre-commit lint hook
- [x] It refactors the lint and prettier flow
- [x] Fix all linting errors
- [x] It adds commit message check hook
- [x] Refactor `.env.example` file
- [x] Add `README.md` file to the repository containing development server setup and env description

## Check list  

- [ ] All Merge Request pipelines should be passed.
- [ ] Make sure all the changes are necessary and correct.
- [ ] Make sure no needed change missing (all necessary changes are made).
- [ ] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [ ] All necessary unit tests are written.
- [ ] Critical changes should be tested locally (or on staging).

### CODE REVIEW

File: ts-version/src/presentation/wss/listeners/orderbook.listener.ts [hunk 1]
```diff
@@ -5,7 +5,10 @@ export default class OrderbookListener implements IEventObserver {
   private static orderbookListener: OrderbookListener;
   private static eventName = "orderbook";
 
-  update(newObject: Observable, oldObject?: Observable) {}
```
Comment (WilliamHawkingStephen) [inline ts-version/src/presentation/wss/listeners/orderbook.listener.ts:9]: its done in another branch : 
https://github.com/dexpresso-protocol/core/pull/374

Comment (GreatGaryCooper) [reply ts-version/src/presentation/wss/listeners/orderbook.listener.ts:9]: Yeah, just wanted to be consistent with places that I added `_` prefix 👍🏻 
Thanks mate 🙏🏻 🌹

File: ts-version/src/data/database/postgres/repository/network.adapter.ts [hunk 2]
```diff
@@ -73,7 +73,7 @@ class NetworkRepoAdapter
     return result as unknown as Network;
   }
 
-  async store(data: INetworkRepo.storeInput): Promise<boolean> {
+  store(data: INetworkRepo.storeInput): boolean {
```
Comment (hypersolo) [inline ts-version/src/data/database/postgres/repository/network.adapter.ts:76]: this function sould be async

File: README.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+**Subject must be one of build, chore, ci, docs, feat, fix, perf, refactor, revert, style, and test.
+**
+
+Examples;
+
+- `docs: add README.md to the project`
+- `feat: add OCO logic to the order service`
+- `fix: connect rate-limiter to the http server as middleware`
+
+_For more information, you can read [this documentation](http://https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional "this documentation")._
```
Comment (hypersolo) [inline README.md]: bad link

File: README.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+| `ZKSYNCV1_MIDDLE_ORDERER_PRIV_KEY`        | A private key for zksyncv1 middle orderer                                                  | `private-key`                                      |
+| `ZKSYNCV1_GOERLI_MIDDLE_ORDERER_PRIV_KEY` | A private key for zksyncv1-goerli middle orderer                                           | `private-key`                                      |
+| `INFURA_PROJECT_IDS`                      | Infura Project IDs(Third Party Node)                                                       | `id1,id2,id3`                                      |
+
+##### VS Code extentions:
+
+- **ESLint: **We use [ESLint](http://https://eslint.org/ "ESLint") as our linter tool, you can lint your code using the provided yarn command down below, but in order to receive real-time feedback as you write your code and handle mistakes as soon as you encounter them, you can use the [VS Code ESLint extention](http://https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint "VS Code ESLint extention").
```
Comment (hypersolo) [inline README.md:68]: bad link

File: README.md (hunk 3 above)
Comment (GreatGaryCooper) [reply README.md]: This one is fixed

File: README.md (hunk 4 above)
Comment (GreatGaryCooper) [reply README.md]: This is fixed now

**Review by hypersolo — APPROVED** [review]

**Pull request merged**